integrations/isaac_ros_3_2_yopo/isaac_ros_yopo_bringup/test/test_relay_source.py

```
@@ -0,0 +1,122 @@
+        with open(LAUNCH_PATH, "r", encoding="utf-8") as stream:
+            cls.launch_source = stream.read()
+        relay_tree = ast.parse(cls.relay_source, filename=RELAY_PATH)
+        finite_function = next(
+            node
+            for node in relay_tree.body
+            if isinstance(node, ast.FunctionDef)
+            and node.name == "imu_measurements_are_finite"
+        )
+        function_module = ast.Module(body=[finite_function], type_ignores=[])
+        namespace = {"Imu": object, "math": math}
+        exec(compile(function_module, RELAY_PATH, "exec"), namespace)
```

Ruff (0.15.21)

[error] Use of `exec` detected

(S102)